tests/phpunit/elementor/modules/components/test-component-lock-manager.php

```diff
@@ -0,0 +1,340 @@
+<?php
+namespace Elementor\Testing\Modules\Components;
+
+use Elementor\Modules\Components\Component_Lock_Manager;
+use Elementor\Modules\Components\Documents\Component as Component_Document;
+use Elementor\Plugin;
+use ElementorEditorTesting\Elementor_Test_Base;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+class Test_Component_Lock_Manager extends Elementor_Test_Base {
+
+	private $lock_manager;
+	private $test_user_1;
+	private $test_user_2;
+	private $test_document_ids;
+
+	public function setUp(): void {
+		parent::setUp();
+
+		// Create test users
+		$this->test_user_1 = $this->factory()->create_and_get_administrator_user()->ID;
+		$this->test_user_2 = $this->factory()->create_and_get_administrator_user()->ID;
+
+		// Create lock manager instance
+		$this->lock_manager = Component_Lock_Manager::get_instance();
+
+		// Register component document type
+		Plugin::$instance->documents->register_document_type(
+			Component_Document::TYPE,
+			Component_Document::get_class_full_name()
+		);
+
+		// Register post type for components
+		register_post_type( Component_Document::TYPE, [
+			'label' => Component_Document::get_title(),
+			'labels' => Component_Document::get_labels(),
+			'public' => false,
+			'supports' => Component_Document::get_supported_features(),
+		] );
+
+		// Create test documents of different types
+		$this->test_document_ids = $this->create_test_documents();
```

Comment:
this naming is a bit strange, because it doesn't hold on document ids, one is of a document, the other is for component.

I'd suggest splitting the two, so that the create test document would just create a document, just like `create_test_component` only creates a component :)

```suggestion
		$this->test_document_id = $this->create_test_document();
		$this->test_component_id = $this->create_test_component();
```